git/object_scanner_test.go

```diff
@@ -228,3 +233,16 @@ func TestObjectScannerWritesResponsesInMultipleChunks(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, []string{"status=success"}, status)
 }
+
+// readRequest preforms a single scan operation on the given `*ObjectScanner`,
+// "s", and returns: an error if there was one, or a request if there was one.
+// If neither, it returns (nil, nil).
+func readRequest(s *ObjectScanner) (*Request, error) {
```

Comment:
How do we detect EOF? That happens when Git exits and closes the pipe. This event should signal LFS to cleanup and exit the process. I think I remember that io.Reader communicates this with an ErrClosedPipe or something?